ehr/sync/task.go

```diff
@@ -34,3 +37,32 @@ func GetClinicId(data map[string]interface{}) (string, error) {
 	}
 	return clinicId, nil
 }
+
+func ScheduleNextExecution(tsk *task.Task) {
+	if tsk.HasError() {
+		tsk.RepeatAvailableAfter(OnErrorAvailableAfterDuration)
+		return
+	}
+
+	cadence := GetCadence(tsk.Data)
+	if cadence == nil {
+		tsk.RepeatAvailableAfter(DefaultCadence)
+	} else if *cadence != 0 {
+		tsk.RepeatAvailableAfter(*cadence)
+	} else {
+		tsk.SetCompleted()
+	}
+}
+
+func GetCadence(data map[string]interface{}) *time.Duration {
+	cadence := data["cadence"].(string)
+	parsed, err := time.ParseDuration(cadence)
+	if err != nil {
+		return nil
```

Comment:
I think it would simplify the logic if instead of returning `nil`, you returned `sync.DefaultCadence`. Both `ScheduleNextExecution` and `GetReconciliationPlan` check if the response is nil, and if so, use the default value.

To go one step further, you could instead return the error (`func GetCandence(...) (time.Duration, error)`) so the caller (who presumably has more context) could decide how to handle that (whether that means using the default, or maybe logging the error before using the default) or maybe using some other value. Just returning nil (or a default) hides the error entirely. If it's really not something we care about, that's fine. If you think there's ever a time you'd want to use that error, consider returning it.

But a minor thing to be sure, and fine how it is.

Reply:
> Both ScheduleNextExecution and GetReconciliationPlan check if the response is nil, and if so, use the default value.

This statement is not true. Planner doesn't use the default value, it check whether the cadence in the task is difference that the one in the settings, and if that's the case it updates the task:
```golang
// Update the task if the configured cadence has been changed
cadenceFromTask := sync.GetCadence(tsk.Data)
if cadenceFromTask == nil || *cadenceFromTask != cadenceFromSettings {
    sync.SetCadence(tsk.Data, cadenceFromSettings)
    update := task.NewTaskUpdate()
    update.Data = &tsk.Data
    toUpdate[tsk.ID] = update
}
```

This function returns exactly what the name implies. If parsing fails, then the cadence is not correctly set so it will be updated as configured. I could return the parsing error, but it will be ignored by the caller.